Sources/Fullscreen/NotificationCenter/PushNotificationDetailsView.swift

```diff
@@ -0,0 +1,148 @@
+}
+
+class PushNotificationDetailsView: UIControl {
+
+    // MARK: - Private properties
+
+    private lazy var magnifyingIconView: UIImageView = {
+        let image = UIImage(named: .magnifyingGlass).withRenderingMode(.alwaysTemplate)
+        let imageView = UIImageView(image: image)
+        imageView.tintColor = .textPrimary
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        imageView.widthAnchor.constraint(equalToConstant: 14).isActive = true
+        imageView.heightAnchor.constraint(equalToConstant: 14).isActive = true
+        return imageView
+    }()
+
+    private lazy var textLabel: Label = {
+        let label = Label(style: .detail, withAutoLayout: true)
+        label.numberOfLines = 1
+        return label
+    }()
+
+    private lazy var textLabelToSuperviewConstraint = textLabel.leadingAnchor.constraint(
+        equalTo: leadingAnchor,
+        constant: .mediumLargeSpacing
+    )
+
+    private lazy var textLabelToMagnifyingIconConstraint = textLabel.leadingAnchor.constraint(
+        equalTo: magnifyingIconView.trailingAnchor,
+        constant: .verySmallSpacing
+    )
+
+    private var highlightedRange: NSRange?
+    private var attributedString: NSMutableAttributedString?
+
+    // MARK: - Init
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Overrides
+
+    override var isHighlighted: Bool {
+        didSet {
+            guard !isHighlighted, oldValue else { return }
+            guard let range = highlightedRange else { return }
+            attributedString?.removeAttribute(.underlineStyle, range: range)
+            attributedString?.addAttribute(.foregroundColor, value: UIColor.textAction, range: range)
+            textLabel.attributedText = attributedString
+        }
+    }
+
+    override func beginTracking(_ touch: UITouch, with event: UIEvent?) -> Bool {
+        guard super.beginTracking(touch, with: event) else { return false }
+        guard let range = highlightedRange else { return false }
+        attributedString?.addAttribute(.underlineStyle, value: NSUnderlineStyle.single.rawValue, range: range)
+        attributedString?.addAttribute(.foregroundColor, value: UIColor.linkButtonHighlightedTextColor, range: range)
+        textLabel.attributedText = attributedString
+        return true
+    }
+
+    // MARK: - Internal methods
+
+    func configure(with details: PushNotificationDetails?) {
+        let textAttributes: [NSAttributedString.Key: Any] = [
+            .font: UIFont.caption,
+            .foregroundColor: UIColor.textPrimary
+        ]
+
+        switch details {
+        case let .link(text, title, showMagnifyingGlass):
+            magnifyingIconView.isHidden = !showMagnifyingGlass
+
+            textLabelToSuperviewConstraint.isActive = magnifyingIconView.isHidden
```

Comment:
If you make this constraint to superview be `lessThanOrEqual` maybe it can always be active?

Reply:
I guess you're right 😉